### PR TITLE
Move sort from YangRang Validate to parseRanges

### PR DESCRIPTION
**Pull request by harshilshah611**
Validate is also called from ygot/ytypes/int_type.go to validate YangRange. This creates race conditions.

### CODE REVIEW

File: pkg/yang/types_builtin.go [hunk 1]
```diff
@@ -859,6 +859,7 @@ func parseRanges(s string, decimal bool, fracDigRequired uint8) (YangRange, erro
 		}
 		r[i] = YRange{min, max}
 	}
+	sort.Sort(r)
 	if err := r.Validate(); err != nil {
 		return nil, err
 	}
```
Comment (wenovus) [inline pkg/yang/types_builtin.go:865]: `coalesce` before `Validate`?

File: pkg/yang/types_builtin.go [hunk 2]
```diff
@@ -944,10 +945,9 @@ func (r YangRange) Less(i, j int) bool {
...
 }
 
-// Validate sorts r and returns an error if r has either an invalid range or has
+// Validate returns an error if r has either an invalid range or has
 // overlapping ranges.
```
Comment (wenovus) [inline pkg/yang/types_builtin.go:949]: Maybe just add at the end of the comment that it expects the range to be sorted.
Maybe also add to the docstring for `parseRanges`.

File: pkg/yang/types_builtin.go [hunk 3]
```diff
@@ -944,10 +945,9 @@ func (r YangRange) Less(i, j int) bool {
...
 func (r YangRange) Validate() error {
-	sort.Sort(r)
```
Comment (wenovus) [inline pkg/yang/types_builtin.go:950]: Hmm, now I'm thinking we should error out if https://golang.org/pkg/sort/#IsSorted returns false.
Also, could you add a `Sort` method for `YangRange`, and call that at `parseRanges`? Could you also add a short test with a couple test cases for the method?

This way if the user somehow manually creates a YangRange, he/she can call `.Sort` to allow it to be validated.

File: pkg/yang/types_builtin.go (hunk 1 above)
Comment (harshilshah611) [reply pkg/yang/types_builtin.go]: I forgot. Done

File: pkg/yang/types_builtin.go (hunk 2 above)
Comment (harshilshah611) [reply pkg/yang/types_builtin.go:948]: Added.

File: pkg/yang/types_builtin.go (hunk 3 above)
Comment (harshilshah611) [reply pkg/yang/types_builtin.go:950]: I see. Done

**Comment by wenovus** [comment]
Thanks, looks good!

**Pull request merged**